KaitaiStream.d.ts

```diff
@@ -0,0 +1,216 @@
+      */
+      readS8le(): number;
+      /**
+        Reads an 8-bit unsigned int from the stream.
+        @return {number} The read number.
+      */
+      readU1(): number;
+      /**
+        Reads a 16-bit big-endian unsigned int from the stream.
+        @return The read number.
+      */
+      readU2be(): number;
+      /**
+        Reads a 32-bit big-endian unsigned int from the stream.
+        @return The read number.
+      */
+      readU4be(): number;
+      /**
+        Reads a 64-bit big-endian unsigned int from the stream. Note that
+        JavaScript does not support 64-bit integers natively, so it will
+        automatically upgrade internal representation to use IEEE 754
+        double precision float.
+        @return The read number.
+      */
+      readU8be(): number;
+      /**
+        Reads a 16-bit little-endian unsigned int from the stream.
+        @return The read number.
+      */
+      readU2le(): number;
+      /**
+        Reads a 32-bit little-endian unsigned int from the stream.
+        @return The read number.
+      */
+      readU4le(): number;
+      /**
+        Reads a 64-bit little-endian unsigned int from the stream. Note that
+        JavaScript does not support 64-bit integers natively, so it will
+        automatically upgrade internal representation to use IEEE 754
+        double precision float.
+        @return The read number.
+      */
+      readU8le(): number;
+      readF4be(): number;
+      readF8be(): number;
+      readF4le(): number;
+      readF8le(): number;
+      alignToByte(): void;
+      readBitsInt(n: number): number;
+      /**
+        Native endianness. Either KaitaiStream.BIG_ENDIAN or KaitaiStream.LITTLE_ENDIAN
+        depending on the platform endianness.
+      */
+      static endianness: boolean;
+      readBytes(len: number): Uint8Array;
+      readBytesFull(): Uint8Array;
+      readBytesTerm(terminator: any, include: any, consume: any, eosError: any): any;
+      ensureFixedContents(expected: any): Uint8Array;
+      static bytesStripRight(data: any, padByte: any): any;
+      static bytesTerminate(data: any, term: any, include: any): any;
+      static bytesToStr(arr: Uint8Array, encoding: string): any;
+      static processXorOne(data: any, key: any): Uint8Array;
+      static processXorMany(data: any, key: any): Uint8Array;
+      static processRotateLeft(data: any, amount: any, groupSize: any): Uint8Array;
+      static processZlib(buf: Uint8Array): any;
+      static mod(a: any, b: any): number;
+      static arrayMin(arr: any): any;
+      static arrayMax(arr: any): any;
+      static byteArrayCompare(a: any, b: any): number;
+      /**
+        Maps a Uint8Array into the KaitaiStream buffer.
+    
+        Nice for quickly reading in data.
+    
+        @param length Number of elements to map.
+        @return Uint8Array to the KaitaiStream backing buffer.
+        */
+      mapUint8Array(length: number): Uint8Array;
```

Comment:
Are these TS definitions supposed to annotate the public interface only? If yes, then these methods (i.e. `mapUint8Array` and everything below) are effectively private and are not supposed to be used from end user's code.

Reply:
TypeScript does have visibility specifiers, and I presume if the whole thing were TypeScript you'd want to mark those methods private. However, since this is just a definition file, practically speaking it really doesn't matter if they're private or omitted. I'll remove them when I get a chance today.

Reply:
Alright, I've removed mapUint8Array and below.